src/Boxer/Platform/Win32/Win32Application.cpp

```diff
@@ -94,42 +191,143 @@ namespace boxer {
+			pfd.cDepthBits = 24;
+			pfd.cStencilBits = 8;
+			pfd.cAuxBuffers = 0;
+			pfd.iLayerType = PFD_MAIN_PLANE;
+			int format = ChoosePixelFormat(win32->hDC, &pfd);
+			SetPixelFormat(win32->hDC, format, &pfd);
+
+			InitGLExt();
+
+			int pixel_format_attribs[] = {
+				WGL_DRAW_TO_WINDOW_ARB,     GL_TRUE,
+				WGL_SUPPORT_OPENGL_ARB,     GL_TRUE,
+				WGL_DOUBLE_BUFFER_ARB,      GL_TRUE,
+				WGL_ACCELERATION_ARB,       WGL_FULL_ACCELERATION_ARB,
+				WGL_PIXEL_TYPE_ARB,         WGL_TYPE_RGBA_ARB,
+				WGL_COLOR_BITS_ARB,         32,
+				WGL_DEPTH_BITS_ARB,         24,
+				WGL_STENCIL_BITS_ARB,       8,
+				0
+			};
+
+			int pixel_format;
+			UINT num_formats;
+			wglChoosePixelFormatARB(win32->hDC, pixel_format_attribs, 0, 1, &pixel_format, &num_formats);
+			if (!num_formats) {
+				ASSERT(0, "Failed to set the OpenGL pixel format.");
+			}
+
+			//PIXELFORMATDESCRIPTOR pfd;
+			DescribePixelFormat(win32->hDC, pixel_format, sizeof(pfd), &pfd);
+			if (!SetPixelFormat(win32->hDC, pixel_format, &pfd)) {
+				ASSERT(0, "Failed to set OpenGL pixel format.");
+			}
 
-		win32->hDC = GetDC(win32->hWnd);
+			int gl33_attribs[] = {
+				WGL_CONTEXT_MAJOR_VERSION_ARB, 3,
+				WGL_CONTEXT_MINOR_VERSION_ARB, 3,
+				WGL_CONTEXT_PROFILE_MASK_ARB,  WGL_CONTEXT_CORE_PROFILE_BIT_ARB,
+				0,
+			};
 
-		int format = ChoosePixelFormat(win32->hDC, &pfd);
-		SetPixelFormat(win32->hDC, format, &pfd);
+			win32->hGL = wglCreateContextAttribsARB(win32->hDC, 0, gl33_attribs);
+			if (!win32->hGL) {
+				ASSERT(0, "Failed to create OpenGL context.");
+			}
 
-		win32->hGL = wglCreateContext(win32->hDC);
-		wglMakeCurrent(win32->hDC, win32->hGL);
+			if (!wglMakeCurrent(win32->hDC, win32->hGL)) {
+				ASSERT(0, "Failed to activate OpenGL context.");
+			}
+		}
 
 		GLenum err = glewInit();
-		if (GLEW_OK != err) {
-			/* Problem: glewInit failed, something is seriously wrong. */
+		if (err != GLEW_OK) {
 			const char* errorStr = (const char*)glewGetErrorString(err);
-			ASSERT(0, errorStr);
+
+			// TODO(NeGate): We need a logging system
+			// TODO(NeGate): We need a logging system
+			// TODO(NeGate): We need a logging system
+			// TODO(NeGate): We need a logging system
+			// TODO(NeGate): We need a logging system
+			// TODO(NeGate): We need a logging system
+			// TODO(NeGate): We need a logging system
+			// TODO(NeGate): We need a logging system
+			// TODO(NeGate): We need a logging system
+			// TODO(NeGate): We need a logging system
+			// TODO(NeGate): We need a logging system
+			// TODO(NeGate): We need a logging system
+			printf("GLEW ERROR: %s\n", errorStr);
 		}
 	}
 
 	void Application::Launch() {
 		Win32Handle* win32 = reinterpret_cast<Win32Handle*>(_Handle);
 
 		// Event handling
-		GLCall(glClearColor(0.0f, 0.0f, 0.0f, 1.0f));
-		while(1) {
-			MSG message;
-			if (PeekMessage(&message, NULL, NULL, NULL, PM_REMOVE) > 0) {
-				TranslateMessage(&message);
-				DispatchMessage(&message);
+		glClearColor(0.0f, 0.0f, 0.0f, 1.0f);

```

Comment:
I usually recommend clearing to something other than pure black or white... like a dark blue. This way if there's ever a shader issue you aren't rendering black-on-black or for some GPUs white-on-white. 
```suggestion
		glClearColor(0.0f, 0.0f, 0.2f, 1.0f);
```